### DISCUSSION TITLE
Form Validate Unique on two Columns?

### DISCUSSION THREAD

**Discussion by yaplej** [filamentphp/filament#2586] [Help] [answered]
Hello,

How can I valid a field to be unique on two columns?

I thought something like this might work.
```
                    ->unique(callback: function (Unique $rule) {
                        return $rule->where('model_id', fn (Closure $get) => $get('model_id'));
                    }),
```

Trying to make the field use the same constraint as the database.
```
$table->unique(['field', 'model_id']);
```

**Comment by Z3d0X** [comment, marked as answer]
You almost got it. `Closure $get` should be at top level closure(this is the closure evaluated by filament) 
```php
->unique(callback: function (Unique $rule, ?Model $record, Closure $get) {
  return $rule->where('model_id', $get('model_id'));
}),
```

**Reply by yaplej** [reply]
That seems to have worked but broke the edit page.  Maybe I need to exclude itself?  That would mean include ignorable: somehow?
https://filamentphp.com/docs/2.x/forms/validation#custom-rules

**Reply by Z3d0X** [reply]
```php
->unique(callback: function (Unique $rule, ?Model $record, Closure $get) {
  return $rule->where('model_id', $get('model_id'))->ignore($record);
}),
```
or
```php
->unique(
  ignoreRecord: true,
  callback: function (Unique $rule,  Closure $get) {
    return $rule->where('model_id', $get('model_id'));
  }
),
```

**Reply by yaplej** [reply]
That worked great!  Thank you.

**Reply by khanakia** [reply]
@danharrin 

I have unique column (company_id, user_name) but when i udpate the user_name and company both together i get the error because the form is not sending the username field

```
update `users` set `company_id` = 1, `users`.`updated_at` = 2022-08-06 12:09:55 where `id` = 3

SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '1-test' for key 'users_company_id_user_name_unique'
```

It should run this query
```update `users` set user_name='test', `company_id` = 1, `users`.`updated_at` = 2022-08-06 12:09:55 where `id` = 3```


I debugged further it seems  form sending multiple update query when **filament select** is used

```

[2022-08-06 12:18:25]
update `users` set `company_id` = ?, `users`.`updated_at` = ? where `id` = ? [1, 2022-08-06 12:18:25, 3]

[2022-08-06 12:18:25]
update `users` set `status` = ?, `user_name` = ?, `hidden` = ?, `is_default` = ?, `users`.`updated_at` = ? where `id` = ? [, test, , , 2022-08-06 12:18:25, 3]
```
